### ISSUE TITLE
Next.js Pages Router results in "MantineProvider was not found in component tree" with v7.1

### ISSUE THREAD

**Issue by daniel-heg** [icflorescu/mantine-datatable#451]
Hi, 
thank you for this library!
When migrating to Mantine v7 I came across an error, which seems to be due to the DataTable component. 

**Describe the bug:**
When installing Mantine-DataTable (v7.1) in a next.js pages environment an error occurs on page load if the DataTable is rendered on the first render. 

**To Reproduce:**
1. Download [minmal-pages-template](https://github.com/mantinedev/next-pages-min-template)
2. `yarn add mantine-datatable`
3. Add layers like described in [Quickstart](https://github.com/icflorescu/mantine-datatable#quickstart)
4. Replace `index.tsx` with [basic example](https://icflorescu.github.io/mantine-datatable/examples/basic-usage/):
```js
/* index.tsx */

import { DataTable } from "mantine-datatable";
import companies from "./companies.json";

export default function IndexPage() {
  return (
      <DataTable
        columns={[{ accessor: 'name' }, { accessor: 'streetAddress' }, { accessor: 'city' }, { accessor: 'state' }]}
        records={companies}
      />
  );
}
```
5. Leads to "Error: @mantine/core: MantineProvider was not found in component tree, make sure you have it in your app":
![Screenshot 2023-11-02 121004](https://github.com/icflorescu/mantine-datatable/assets/50105152/034dc7d8-a4a3-4590-8b4a-179aae8f07bc)

 
**Expected behavior:**
Obviously the MantineProvider is added in `_app.tsx` and I would expect the page to render correctly. 
Interesingly there is no error if the DataTable is rendered later, after the page has loaded. 
So I suspect that some kind of dynamic import is not working?

**Desktop:**
- OS: Windows
- Browser Firefox

**Comment by icflorescu** [comment]
That's a bummer. I spent most of yesterday investigating this.

It only happens with Next.js **pages router** and I didn't catch it before because the new documentation site is using the app router.

So far, I've managed to track it down to the fact that somehow, Next.js will load the `cjs` version of the library, resulting in two instances of Mantine. Kind of similar to the issue depicted in [this discussion](https://github.com/orgs/mantinedev/discussions/4718), though we're not using a monorepo configuration, and we're not bundling anything.

I noticed this doesn't happen if I only export `esm`. In other words, changing the `package.json` from this:

```js
{
  // ...
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/index.js",
  "types": "./dist/types/index.d.ts",
  "exports": {
    ".": {
      "import": {
        "types": "./dist/types/index.d.ts",
        "default": "./dist/esm/index.js"
      },
      "require": {
        "types": "./dist/types/index.d.ts",
        "default": "./dist/cjs/index.js"
      }
    },
    "./styles.css": "./dist/css/styles.css",
    "./styles.layer.css": "./dist/css/styles.layer.css"
  },
  // ...
}
```

...to this:

```js
{
  // ...
  "types": "./dist/types/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/esm/index.js"
    },
    "./styles.css": "./dist/css/styles.css",
    "./styles.layer.css": "./dist/css/styles.layer.css"
  },
  // ...
}
```

...would make the issue go away.

I wonder if I could switch to `esm`-only packaging. I mean, are there still scenarios when `mantine-datatable` is used with `cjs`?...

Since a lot of people will definitely want to use it with the Next.js pages router, I think I'm going to risk it and see what happens.

What do you think?

**Comment by daniel-heg** [comment]
Thank you for your extensive investigation. 

I also tried to (re-)configure my `next.config.js` with this new information, but nothing really seems to work. The closest I got to making it work is when I add `mantine-datatable` to `transpilePackages` in the config. But then I get hydration errors.

I personally have not used `CommonJS` in some time and I also would not use it in any new projects. 
I can only imagine that a minority still requires `CJS`, so in my opion the support could be dropped. Especially since (to my knowledge) other packages also started dropping support. 

However, since Mantine itself still supports `CJS`, I guess that retaining that compatibility would be nice. 

Sadly I am not really knowledgeable in building packages, so I am not sure what else there could be done. I tried getting inspired by the [@mantine/core](https://github.com/mantinedev/mantine/blob/master/src/mantine-core/package.json) package file. The only noticeable difference to me is that in that case `.mjs` files are used. Maybe next.js preferes files with that extension?
